### PR TITLE
8283245: Create a test for JDK-4670319

### PR DESCRIPTION
**Pull request by srmandal**
Write an automated regression test for  JDK-4670319

Issue
When a JTree node is expanded or collapsed, an Accessibility PropertyChange event is fired with the old state of "collapsed" and new state of "expanded" (or vice versa). The problem is that the source of the event is the AccessibeJTree, and not the AccessibleJTreeNode. This means that an assistive technology listening to this event is unable to report to the user what node was expanded/collapsed.

Fix
Fix for [JDK-4670319](https://bugs.openjdk.java.net/browse/JDK-4670319) addresses the above issue.  When tree node is expanded/collapsed, PropertyChangeEventSource is the Node.
This review is for a test for validating the above issue.

This review is for migrating tests from a closed test suite to open.

Testing:
The test ran successfully on Mach5 with multiple runs (30) on windows-x64, linux-x64 and macos-x64.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283245](https://bugs.openjdk.java.net/browse/JDK-8283245): Create a test for JDK-4670319

### Reviewers
 * [Sergey Bylokhov](https://openjdk.java.net/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8012/head:pull/8012` \
`$ git checkout pull/8012`

Update a local copy of the PR: \
`$ git checkout pull/8012` \
`$ git pull https://git.openjdk.java.net/jdk pull/8012/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8012`

View PR using the GUI difftool: \
`$ git pr show -t 8012`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8012.diff">https://git.openjdk.java.net/jdk/pull/8012.diff</a>

</details>

### CODE REVIEW

File: test/jdk/javax/accessibility/4670319/AccessibleJTrePCESourceTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @key headful
+ * @bug 4670319
+ * @summary AccessibleJTree should fire a PropertyChangeEvent using a AccessibleJTreeNode as source.
```
Comment (mrserb) [inline test/jdk/javax/accessibility/4670319/AccessibleJTrePCESourceTest.java:28]: Please split the long lines, use the 80 chars per line.

File: test/jdk/javax/accessibility/4670319/AccessibleJTrePCESourceTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    private static JTree jTree;
+    private static JFrame jFrame;
+
+    private static ArrayList<PropertyChangeEvent> eventsList = new ArrayList<PropertyChangeEvent>();
+
+    private static void doTest() throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> createGUI());
+            Robot robot = new Robot();
+            robot.setAutoDelay(500);
```
Comment (mrserb) [inline test/jdk/javax/accessibility/4670319/AccessibleJTrePCESourceTest.java:50]: Are you sure that the autodelay is needed here? The test does not use any method other than `waitForIdle`

**Review by mrserb — APPROVED** [review]

**Comment by srmandal** [comment]
@mrserb Can you please sponsor this PR?

**Pull request closed without merging**